pkg/diagnose/diagnose.go

```diff
@@ -144,6 +144,15 @@ func (c *Collector) CollectPodLogs(corePodSelector labels.Selector) {
 // ExportDiagnostics info
 func (c *Collector) ExportDiagnostics(destDir string) {
 	targetFile := fmt.Sprintf("%s.tar.gz", c.folderName)
+	if destDir != "" {
+		if _, err := os.Stat(destDir); os.IsNotExist(err) {
+			err := os.MkdirAll(destDir, os.ModePerm)
+			if err != nil {
+				c.log.Fatalf(`❌ Could not create directory %s, reason: %s`, destDir, err)
+			}
+		}
+		targetFile = fmt.Sprintf("%s/%s", destDir, targetFile)
```

Comment:
Make sure that writing the TAR from one file system to the second one works.

Reply:
Pulkit is checking it on the same scenario, but there is no call to os.rename in the Tar function.